rosbridge_library/src/rosbridge_library/internal/message_conversion.py

```diff
@@ -290,28 +326,37 @@ def _to_inst(msg, rostype, roottype, clock=None, inst=None, stack=None):
     # Check if it's uint8[], and if it's a string, try to b64decode
     for binary_type, expression in ros_binary_types_list_braces:
         if expression.sub(binary_type, rostype) in ros_binary_types:
+            assert isinstance(msg, Sequence)
             return _to_binary_inst(msg)
 
     # Check the type for time or rostime
     if rostype in ros_time_types:
+        assert isinstance(msg, dict | str)
+        assert inst is None or isinstance(inst, TimeMsg | DurationMsg), (
+            "Expected Time or Duration instance"
+        )
```

Comment:
I think asserts are not favorable, as they will just raise an `AssertionError` instead of manually raising a `TypeError`.